src/genpy/generator.py

```diff
@@ -772,7 +772,7 @@ def msg_generator(msg_context, spec, search_path):
     
     full_text = compute_full_text_escaped(msg_context, spec)
     # escape trailing double-quote, unless already escaped, before wrapping in """
-    if full_text[-1] == '"' and not full_text[-2:] == r'\"':
+    if len(full_text) > 0 and full_text[-1] == '"' and not full_text[-2:] == r'\"':
```

Comment:
I thought about something similar yesterday :wink: but only tried it with a string of length 1 which the list slicing if obviously fine with...

Anyway I think we should make it even more simple and defensive now:

    if full_text.endswith('"') and not full_text.endswith(r'\"'):